nucypher/blockchain/eth/actors.py

```diff
@@ -336,6 +320,54 @@ def __init__(
             ThresholdRequestDecryptingPower
         )  # used for secure decryption request channel
 
+        self.condition_providers = self.connect_condition_providers(
+            condition_provider_uris
+        )
+
+    def connect_condition_providers(
+        self, condition_provider_uris: Optional[Dict[int, List[str]]] = None
+    ) -> DefaultDict[int, Set[HTTPProvider]]:
+        """Multi-provider support"""
+
+        # If condition_provider_uris is `None` the node operator
+        # did not configure any additional condition providers.
+        condition_provider_uris = condition_provider_uris or dict()
+
+        # These are the chains that the Ritualist will connect to for conditions evaluation (read-only).
+        condition_providers = defaultdict(set)
+
+        # Borrow default RPC endpoints from the Operator.
+        eth_chain = self.application_agent.blockchain
+        polygon_chain = self.payment_method.agent.blockchain
+        condition_providers[eth_chain.client.chain_id].add(eth_chain.provider)
+        condition_providers[polygon_chain.client.chain_id].add(polygon_chain.provider)
+
+        # Add any additional providers that were passed in.
+        for chain_id, condition_provider_uris in condition_provider_uris.items():
+            if chain_id not in _CONDITION_CHAINS:
+                # this is a safety check to prevent the Ritualist from connecting to
+                # chains that are not supported by ursulas on the network;
+                # Prevents the Ursula/Ritualist from starting up if this happens.
+                raise NotImplementedError(
+                    f"Chain ID {chain_id} is not supported by the Ritualist."
+                )
+
+            providers = list()
+            for uri in condition_provider_uris:
+                provider = HTTPProvider(endpoint_uri=uri)
+                providers.append(provider)
+            condition_providers[chain_id].update(providers)
```

Comment:
What if the payment_method and the condition_provider_uris used/had different endpoint URIs for the same chain? Des one get overwritten by the other? Not that this would typically happen of course 😅 .

Kind of related to my earlier comment in this file.

Reply:
Nope they will not be overwritten, just maintained as separate endpoints.